packages/datadog-instrumentations/src/playwright.js

```diff
@@ -93,13 +99,56 @@ function getProjectsFromDispatcher (dispatcher) {
   return dispatcher._loader?.fullConfig()?.projects
 }
 
-function getBrowserNameFromProjects (projects, projectId) {
-  if (!projects) {
+function getBrowserNameFromProjects (projects, test) {
+  if (!projects || !test) {
     return null
   }
-  return projects.find(project =>
-    project.__projectId === projectId || project._id === projectId
-  )?.name
+  const { _projectIndex, _projectId: testProjectId } = test
+
+  if (_projectIndex !== undefined) {
+    return projects[_projectIndex]?.name
+  }
+
+  return projects.find(({ __projectId, _id, name }) => {
+    if (__projectId !== undefined) {
+      return __projectId === testProjectId
+    }
+    if (_id !== undefined) {
+      return _id === testProjectId
+    }
+    return name === testProjectId
+  })?.name
+}
+
+function formatTestHookError (error, hookType, isTimeout) {
+  let hookError = error
+  if (error) {
+    hookError = error
+    hookError.message = `Error in ${hookType} hook: ${error.message}`
+  }
+  if (!hookError && isTimeout) {
+    hookError = new Error(`${hookType} hook timed out`)
+  }
+  return hookError
+}
+
+function addErrorToTestSuite (testSuiteAbsolutePath, error) {
+  if (!testSuiteToErrors.has(testSuiteAbsolutePath)) {
+    testSuiteToErrors.set(testSuiteAbsolutePath, [error])
+  } else {
+    testSuiteToErrors.get(testSuiteAbsolutePath).push(error)
+  }
+}
```

Comment:
pet peeve nit: I think this reads more straightforward:
```suggestion
function addErrorToTestSuite (testSuiteAbsolutePath, error) {
  if (testSuiteToErrors.has(testSuiteAbsolutePath)) {
    testSuiteToErrors.get(testSuiteAbsolutePath).push(error)
  } else {
    testSuiteToErrors.set(testSuiteAbsolutePath, [error])
  }
}
```
"Just say no to `!`."

Reply:
agreed, thanks